util/run.bzl

```diff
@@ -0,0 +1,199 @@
+
+    return struct(runfiles=ctx.runfiles(files = [
+            ctx.executable.image_tar,
+            ctx.file.image_tar] + 
+            ctx.attr.image_tar.files.to_list() + 
+            ctx.attr.image_tar.data_runfiles.files.to_list()
+        ),
+    )
+
+_run_and_commit = rule(
+    attrs = {
+        "flags": attr.string_list(
+            doc = "list of flags to pass to run command",
+            default = [],
+        ),
+        "image_tar": attr.label(
+            executable = True,
+            allow_files = True,
+            mandatory = True,
+            single_file = True,
+            cfg = "target",
+        ),
+        "original_image": attr.string(
+            doc = "name of original image (for computing sha)",
+            mandatory = True,
+        ),
+        "image_name": attr.string(
+            doc = "name of image to run commands on",
+            mandatory = True,
+        ),
+        "command": attr.string_list(
+            doc = "command to run",
+            mandatory = True,
+            non_empty = True,
+        ),
+        "_run_tpl": attr.label(
+            default = Label("//util:commit.sh.tpl"),
+            allow_files = True,
+            single_file = True,
+        ),
+    },
+    executable = True,
+    implementation = _commit_impl,
+)
+
+_run_and_extract = rule(
+    attrs = {
+        "flags": attr.string_list(
+            doc = "list of flags to pass to run command",
+            default = [],
+        ),
+        "image_tar": attr.label(
+            executable = True,
+            allow_files = True,
+            mandatory = True,
+            single_file = True,
+            cfg = "target",
+        ),
+        "image_name": attr.string(
+            doc = "name of image to run commands on",
+            mandatory = True,
+        ),
+        "command": attr.string_list(
+            doc = "command to run",
+            mandatory = True,
+            non_empty = True,
+        ),
+        "extract_file": attr.string(
+            doc = "path to file to extract from container",
+            mandatory = True,
+        ),
+        "target": attr.string(
+            doc = "path to extract file to on host machine",
+            default = ".",
+        ),
+        "_extract_tpl": attr.label(
+            default = Label("//util:extract.sh.tpl"),
+            allow_files = True,
+            single_file = True,
+        ),
+    },
+    executable = True,
+    implementation = _extract_impl,
+)
+
+def container_run_and_commit(name, image, command, flags=None):
+    image_tar, intermediate_image = _rename_image(image, name)
+
+    _run_and_commit(
+        name = name,
+        original_image = image,
+        image_name = intermediate_image,
+        image_tar = image_tar + ".tar",
+        flags = flags,
+        command = command,
+    )
+
+def container_run_and_extract(name, image, command, extract_file, target, flags=None):
+    image_tar, intermediate_image = _rename_image(image, name)
```

Comment:
Add docs in format mentioned here https://skydoc.bazel.build/docs/writing.html so skydoc can pick it up.

e.g.:  https://github.com/GoogleCloudPlatform/base-images-docker/blob/master/package_managers/download_pkgs.bzl#L86